multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UnbindServiceFromApplicationStep.java

```diff
@@ -37,4 +39,41 @@ protected String getStepErrorMessage(ProcessContext context) {
                                                                                                   .getName());
     }
 
+    private ApplicationServicesUpdateCallback getApplicationServicesUpdateCallback(ProcessContext context) {
+        return new UnbindServiceFromApplicationCallback(context);
+    }
+
+    public static class UnbindServiceFromApplicationCallback implements ApplicationServicesUpdateCallback {
+
+        private final ProcessContext context;
+
+        public UnbindServiceFromApplicationCallback(ProcessContext context) {
+            this.context = context;
+        }
+
+        @Override
+        public void onError(CloudOperationException e, String applicationName, String serviceName) {
+            context.getStepLogger()
+                   .infoWithoutProgressMessage(Messages.CHECK_IF_SERVICE_BINDING_BETWEEN_SERVICE_0_AND_APP_1_ALREADY_DELETED, serviceName,
+                                               applicationName);
+            CloudControllerClient client = context.getControllerClient();
+
+            try {
+                if (ServiceBindingUtil.getServiceBinding(client, applicationName, serviceName) == null) {
+                    context.getStepLogger()
+                           .infoWithoutProgressMessage(Messages.SERVICE_BINDING_BETWEEN_SERVICE_0_AND_APP_1_ALREADY_DELETED, serviceName,
+                                                       applicationName);
+                    return;
+                }
+            } catch (CloudOperationException e2) {
+                context.getStepLogger()
+                       .infoWithoutProgressMessage(Messages.COULD_NOT_DETECT_IF_SERVICE_BINDING_BETWEEN_SERVICE_0_AND_APP_1_ALREADY_DELETED,
```

Comment:
Maybe use: errorWithoutProgressMessage

Reply:
Thanks, it will be removed because we will not perform additional call to get bindings